drpg/sync.py

```diff
@@ -100,30 +103,55 @@ def _process_item(self, product: Product, item: DownloadItem) -> None:
                 },
             )
 
-            path.parent.mkdir(parents=True, exist_ok=True)
-            path.write_bytes(file_response.content)
+            if (
+                self._validate
+                and (checksum := _newest_checksum(item))
+                and md5(file_response.content).hexdigest() != checksum
```

Comment:
Something is off, some files always have mismatched checksums

Reply:
This may be related: https://github.com/glujan/drpg/issues/118#issuecomment-2558434849